src/Tokenizer/Analyzer/Analysis/TypeAnalysis.php

```diff
@@ -12,50 +12,53 @@
 
 namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
 
-/**
- * @internal
- */
-final class FunctionReturnTypeAnalysis implements StartEndTokenAwareAnalysis
+class TypeAnalysis implements StartEndTokenAwareAnalysis
 {
+    private static $scalarTypes = [
+        'array',
+        'bool',
+        'callable',
+        'int',
+        'iteratable',
+        'float',
+        'string',
```

Comment:
http://php.net/manual/pl/function.is-scalar.php

scalar in PHP has a meaning already, reusing scalar word with different definition what is a scalar type actually gonna be super misleading

Reply:
This is rather to exclude PHP-reserved types to avoid bugs.

I got a fixer that changes varaible name to it's type.
And without `mixed`/`object` as scalar (or *PHP-reserved* types) it made mess like:

```diff
 /**
  * @var mixed
  */
-private $variable;
+private $mixed;
```

Reply:
i'm not saying anything about **calling** the `_is_scalar`.
i'm saying that scalar term already have a meaning in PHP as pseudotype, and it doesn't cover objects. reusing the term but with different meaning is super misleading

Reply:
What term would you suggest?

Reply:
change used term to some non-established in php internals and document what it is.
for now, i must admit that i have no clue what is this info suppose to be.
reverse of collection? but then why not `resource` ? no clue

Reply:
@veewee Nice, thanks for the references in comments!